### ISSUE TITLE
If MET workflow fails for the right reason we need to handle it correctly

### ISSUE THREAD

**Issue by tonygardella** [PecanProject/pecan#1327]
Sometimes met workflow will cause failures down the line even if it is doing its job.

## Description
Sometimes gaps are too large or an empty met file is created, but a dbfile record is created which will cause runs to fail. Also, there are times when data isn't available for a specific data/site.  We need to have some checks and/or alternatives in place to alleviate these issues and make it clear to a user what they should do (ex. pick a different met product, year, or date)

## Possible Implementation
@serbinsh mentioned the possibility of substituting in or suggesting the use of other met sources when these issues happen.

**Comment by serbinsh** [comment]
Possibly even allowing you to restart the workflow from the met process with a new selection (e.g. CRUNCEP) if, say AmerifluxLBL fails during gapfill where too much missing data causes a fail, without just giving you a suggestion but failing requiring you to start from scratch.



On Apr 6, 2017, at 2:10 PM, Anthony Gardella <notifications@github.com<mailto:notifications@github.com>> wrote:


Sometimes met workflow will cause failures down the line even if it is doing its job.

Description

Sometimes gaps are too large or an empty met file is created, but a dbfile record is created which will cause runs to fail. Also, there are times when data isn't available for a specific data/site. We need to have some checks and/or alternatives in place to alleviate these issues and make it clear to a user what they should do (ex. pick a different met product, year, or date)

Possible Implementation

@serbinsh<https://github.com/serbinsh> mentioned the possibility of substituting in or suggesting the use of other met sources when these issues happen.

**Comment by mdietze** [comment]
I totally understand where the request for this is coming from, but if you start to think about implementation the number of steps to do these things would be major and require major changes to the workflow. Is this really where we want to devote months of work? Making it so PEcAn would know before it does something that it's going to fail is a pretty hard problem. Likewise, jumping from a failed workflow.R back into the start of the webpage system, but remembering and setting all the variables, is also pretty hard.

Rather than envisioning an ideal system, would there be low-cost & high priority first starts? To me making sure we don't insert failed or empty records into the database seems like a simple LOCAL first start.  The other first start would be a more clear message in the web that products may not be available and documentation about what to do when met fails.  

I also want to make it clear than any changes have to be general and extensible -- we can't have error messages that are model or product specific hacks.

**Comment by serbinsh** [comment]
@mdietze Fair enough on some of the points but I also think that there are still low hanging fruit, you touched on some. 

For example:
1) Detect empty or failed files so we don't skip MET process and then fail on model execution
2) Better error messages rather than just saying things like "gap fill of XXX failed".  Some folks probably wont know what that means or what to do....instead if a met driver can't be created can we be more clear so a user can make another selection; without being too specific?  Then stop and let them start over again.  Just as you mention in you comments.  Probably solve much of this with expanded documentation 

The [much] harder one 
3) Fail, hold and allow for another selection during workflow.......I don't think this requires knowing anything ahead of time...but that is a very high level thought without too much thinking about specifics.

**Comment by serbinsh** [comment]
But again, there are simpler ones to address that we could take care of and then keep the blue sky for another time.
